ol-keycloak/ol-spi/src/main/java/authentication/authenticators/browser/UsernamePasswordForm.java

```diff
@@ -0,0 +1,32 @@
+package authentication.authenticators.browser;
+
+import org.keycloak.authentication.AuthenticationFlowContext;
+import org.keycloak.forms.login.LoginFormsProvider;
+import org.keycloak.forms.login.freemarker.LoginFormsUtil;
+import org.keycloak.forms.login.freemarker.model.IdentityProviderBean;
+import org.keycloak.models.IdentityProviderModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+
+import java.net.URI;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class UsernamePasswordForm extends org.keycloak.authentication.authenticators.browser.UsernamePasswordForm {
+
+    @Override
+    public void authenticate(AuthenticationFlowContext context) {
+        if (context.getUser() != null) {
+            final RealmModel realm = context.getRealm();
```

Comment:
What happens if a user does not have an account and types in their username, what Realm will this return?

Reply:
The user will not reach this screen with the flow shown in the PR description